### PR TITLE
[FLINK-2608] Updated Twitter Chill version.

### PR DESCRIPTION
**Pull request by chermenin**
Fixed JIRA issue [[FLINK-2608] Arrays.asList(..) does not work with CollectionInputFormat](https://issues.apache.org/jira/browse/FLINK-2608) as issue in Twitter Chill project (https://github.com/twitter/chill/issues/255). Updated version of Twitter Chill in `pom.xml`.

### CODE REVIEW

**Comment by zentol** [comment]
Could you include a simple test to verify that this does in fact fix the issue?

**Comment by rmetzger** [comment]
Does this change have any implications on our Kryo version, since Chill since 0.8.0 uses Kryo3 ? https://github.com/twitter/chill/releases/tag/0.8.0

**Comment by chermenin** [comment]
@rmetzger If I'm not mistaken Kryo 2.24 and 3.0 versions is compatible each other at the level of standard IO serialization. And if it possible we will be able to a little wait for Chill 0.7.5 version (with needed fixes and Kryo 2.x) or migrate Flink to Kryo 3.x version.

**Comment by fhueske** [comment]
Bumping the version of serializers is a sensitive issue in Flink. 
If the serialization format changes, savepoints of previous versions can no longer be used.

**Comment by rmetzger** [comment]
This change adds Kryo-shaded to our dependency tree:

```
[INFO] |  +- com.twitter:chill_2.10:jar:0.8.1:compile
[INFO] |  |  +- com.twitter:chill-java:jar:0.8.1:compile
[INFO] |  |  \- com.esotericsoftware:kryo-shaded:jar:3.0.3:compile
[INFO] |  |     \- com.esotericsoftware:minlog:jar:1.3.0:compile
```
I suspect that maven is not recognizing this because chill seems to depend on `kryo-shaded`.
Apparently, `kryo-shaded` has a shaded ASM version included, but it is not relocating the regular Kryo classes. So we'll end up having two Kryo versions in our classpath.
So if we want to upgrade Kryo, we need to do it explicitly, to avoid having two Kryo versions in our classpath.

Another issue we need to consider is the serialization compatibility. Savepoints in Flink could potentially contain data serialized with Kryo 2.24. If we want to provide savepoint compatibility between Flink 1.1 and 1.2, we need to consider that.
According to the Kryo documentation, 2.24 to 3.0.0 is serialization compatible (I hope the same holds true for chill): https://github.com/EsotericSoftware/kryo/blob/master/CHANGES.md#2240---300-2014-10-04
I would like to hear @StephanEwen and @uce's opinion on this.

**Comment by StephanEwen** [comment]
As far as I read it, Kryo 3.x is not strictly serialization compatible with 2.x, hence the major version number bump.

If the interfaces are still stable, then it should be fine to bump the chill dependency version, exclude any kryo dependency, and add our own 2.x kryo dependency. I would prefer that approach.

**Comment by StephanEwen** [comment]
Okay, in detail it seems that compatibility is not maintained for "Unsafe based I/O", but maintained for standard I/O.

**Comment by StephanEwen** [comment]
@chermenin Do you plan to continue with this pull request? Would be great if you

I would suggest to adjust it in the following way:
  - Update the Chill dependency (validate if they change any existing serializer)
  - Exclude Kryo version 2 and version 3 from the Chill dependency. That way, Flink's Kryo version should be used by Chill as well.

Also, testing the serializers via a "GroupReduce" integration test seems heavy and unspecific. Can you change this to a serializer Unit test? Have a look at the Kryo Serializer unit tests for an example.

Do you think that would work?

**Comment by chermenin** [comment]
I exclude Kryo as a dependency from Chill and add simple test to check Java collections. Tests added to `flink-tests` because it is depends on `flink-runtime` (where used Chill).

File: flink-tests/src/test/java/org/apache/flink/test/javaApiOperators/util/CollectionDataSets.java [hunk 1]
```diff
@@ -710,7 +712,103 @@ public String toString() {
...
+		pwc2.sqlDate = new java.sql.Date(gcl2.getTimeInMillis()); // 1976
+
+
+		data.add(pwc1);
+		data.add(pwc2);
+
+		return env.fromCollection(data);
+	}
+
+	public static DataSet<PojoWithCollection> getPojoWithArraysAsListCollection(ExecutionEnvironment env) {
```
Comment (StephanEwen) [inline flink-tests/src/test/java/org/apache/flink/test/javaApiOperators/util/CollectionDataSets.java]: Are these still needed any more by the tests?

**Comment by StephanEwen** [comment]
I like this approach. One minor question, otherwise +1 to go here

File: flink-tests/src/test/java/org/apache/flink/test/javaApiOperators/util/CollectionDataSets.java (hunk 1 above)
Comment (chermenin) [inline flink-tests/src/test/java/org/apache/flink/test/javaApiOperators/util/CollectionDataSets.java]: I think it can be left. Just on the safe side.

**Comment by chermenin** [comment]
@StephanEwen What about merging this PR?

**Comment by mxm** [comment]
The build fails during the Maven Rat plugin license check. Could you fix the build and rebase to the latest master? I think we can merge then.

**Comment by chermenin** [comment]
Yep, fixed.

**Comment by mxm** [comment]
Thank you!

**Comment by StephanEwen** [comment]
@mxm @chermenin Can we please do a followup to this, given that it is already merged?

The tests in `GroupReduceITCase` need to be replaced by proper targeted unit tests.

Sorry for being super strict, but our build times are already exploding, and it is exactly cases like this one (adding distributed runtime Integration Tests to test a serializer) that contribute a lot to that.

**Comment by mxm** [comment]
Sure, I'll take care of it. I saw you giving a +1 and didn't see an issue myself.

**Pull request closed without merging**